### ISSUE TITLE
Download file with encryption

### ISSUE THREAD

**Issue by Kyaw-Zin-Thant** [fluttercommunity/flutter_downloader#128]
I want to download file that can only view in my flutter application.

**Comment by hnvn** [comment]
I don't get your point, what's your use-case?

**Comment by Kyaw-Zin-Thant** [comment]
I developed flutter application that can download video file and display. But downloaded video file from my application can be viewed in gallery. I don't want to view downloaded video file in other application. How can I do?

**Comment by yorickreum** [comment]
Same problem! Works fine on iOS by downloading to ApplicationDocumentsDirectory. But sadly, this doesn't seem to be possible on Android…

**Comment by guangyingmodao** [comment]
> I developed flutter application that can download video file and display. But downloaded video file from my application can be viewed in gallery. I don't want to view downloaded video file in other application. How can I do? @hnvn

1.2.0 - 27.08.2019 #
Android: support FlutterFragmentActivity, fix bug downloaded image/video files not shown in Gallery application, improved HTTP redirection implementation, fix bug cannot open apk file in some cases

maybe you can try 1.1.9 or 1.1.8

**Comment by PannagaShetty** [comment]
i have the same problem did any one get the answer

**Comment by deepashree-ballodi** [comment]
is anyone got the solution how to encrypt video before downloading.

**Comment by PannagaShetty** [comment]
> is anyone got the solution how to encrypt video before downloading.

yeah remove the file extention and save in the obb folder so that the video wont be shown and cant be read by mx player also.

**Comment by deepashree-ballodi** [comment]
> > is anyone got the solution how to encrypt video before downloading.
> 
> yeah remove the file extention and save in the obb folder so that the video wont be shown and cant be read by mx player also.

I want to encrypt whole video and save it locally when user want to watch video offline I need to decrypt and play video from my local encrypted folder. This is what I wanted to do.

**Comment by PannagaShetty** [comment]
> > > is anyone got the solution how to encrypt video before downloading.
> > 
> > 
> > yeah remove the file extention and save in the obb folder so that the video wont be shown and cant be read by mx player also.
> 
> Thanks for the reply PannagaShetty, but I want to encrypt whole video and save it locally when user want to watch video offline I need to decrypt and play video from my local encrypted folder. This is what I wanted to do.

yeah i had a same requirement but thing is for that you have to have your own file extention which can be on a harder side.

**Comment by Elias8** [comment]
Any update?

**Comment by uc-dve** [comment]
> > is anyone got the solution how to encrypt video before downloading.
> 
> yeah remove the file extention and save in the obb folder so that the video wont be shown and cant be read by mx player also.

I have tried. but how we show the video without extension, or how we add extension in flutter after getting the File.

**Comment by PannagaShetty** [comment]
Just open the video file in the video player plugin without extension it
will work it is working for me

**Comment by SaliBhdr** [comment]
Any updates here?

**Comment by EngineerDanny** [comment]
Any updates?

**Comment by anaseem80** [comment]
Any updates?
